huaweicloud/services/acceptance/cfw/resource_huaweicloud_cfw_black_white_list_test.go

```diff
@@ -0,0 +1,175 @@
+	)
+	getBlackWhiteListClient, err := cfg.NewServiceClient(getBlackWhiteListProduct, region)
+	if err != nil {
+		return nil, fmt.Errorf("error creating CFW Client: %s", err)
+	}
+
+	getBlackWhiteListPath := getBlackWhiteListClient.Endpoint + getBlackWhiteListHttpUrl
+	getBlackWhiteListPath = strings.ReplaceAll(getBlackWhiteListPath, "{project_id}", getBlackWhiteListClient.ProjectID)
+
+	getBlackWhiteListqueryParams := buildGetBlackWhiteListQueryParams(state)
+	getBlackWhiteListPath += getBlackWhiteListqueryParams
+
+	getBlackWhiteListOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			200,
+		},
+	}
+	getBlackWhiteListResp, err := getBlackWhiteListClient.Request("GET", getBlackWhiteListPath, &getBlackWhiteListOpt)
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving BlackWhiteList: %s", err)
+	}
+
+	getBlackWhiteListRespBody, err := utils.FlattenResponse(getBlackWhiteListResp)
+	if err != nil {
+		return nil, err
+	}
+
+	lists, err := jmespath.Search("data.records", getBlackWhiteListRespBody)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing data.records from response= %#v", getBlackWhiteListRespBody)
+	}
+
+	if len(lists.([]interface{})) != 1 {
+		return nil, golangsdk.ErrDefault404{}
+	}
+
+	return lists.([]interface{})[0], nil
+}
+
+func TestAccBlackWhiteList_basic(t *testing.T) {
+	var obj interface{}
+
+	rName := "huaweicloud_cfw_black_white_list.test"
+
+	rc := acceptance.InitResourceCheck(
+		rName,
+		&obj,
+		getBlackWhiteListResourceFunc,
+	)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { acceptance.TestAccPreCheck(t) },
+		ProviderFactories: acceptance.TestAccProviderFactories,
+		CheckDestroy:      rc.CheckResourceDestroy(),
+		Steps: []resource.TestStep{
+			{
+				Config: testBlackWhiteList_basic(),
+				Check: resource.ComposeTestCheckFunc(
+					rc.CheckResourceExists(),
+					resource.TestCheckResourceAttr(rName, "list_type", "4"),
+					resource.TestCheckResourceAttr(rName, "direction", "0"),
+					resource.TestCheckResourceAttr(rName, "protocol", "6"),
+					resource.TestCheckResourceAttr(rName, "port", "22"),
+					resource.TestCheckResourceAttr(rName, "address_type", "0"),
+					resource.TestCheckResourceAttr(rName, "address", "1.1.1.1"),
+				),
+			},
+			{
+				Config: testBlackWhiteList_basic_update(),
+				Check: resource.ComposeTestCheckFunc(
+					rc.CheckResourceExists(),
+					resource.TestCheckResourceAttr(rName, "list_type", "4"),
+					resource.TestCheckResourceAttr(rName, "direction", "1"),
+					resource.TestCheckResourceAttr(rName, "protocol", "-1"),
+					resource.TestCheckResourceAttr(rName, "port", "80"),
+					resource.TestCheckResourceAttr(rName, "address_type", "0"),
+					resource.TestCheckResourceAttr(rName, "address", "2.2.2.2"),
+				),
+			},
+			{
+				ResourceName:      rName,
+				ImportState:       true,
+				ImportStateVerify: true,
+				ImportStateIdFunc: testBlackWhiteListImportState(rName),
```

Comment:
Doesn't `object_id` need to be ignored when importing resource?
What's difference between `address_group` and `black_white_list`.

Reply:
we don't need to ignore the `object_id`, because needed to import the list, and inputted by the use, in other words, it won't be empty.
in `address_group`, `object_id` only needed in the creating, and won't be returned by API, so it need to be ignored